### PR TITLE
items: implement detail view

### PR DESCRIPTION
**Pull request by Garfield-fr**
## Why are you opening this PR?

- Task 1152: implement item detail view

## How to test?

*On REROI LS*
- Checkout branch dev
- `script/bootstrap`
- `script/setup`

*On RERO ILS UI*
- Go to document view (Ex: Highly important snuff boxes : the property of the British Rail Pension Fund : Sotheby's, Geneva, 15 May 1990)
- Click on item barcode to show detail view

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+})
+export class ItemTransactionComponent {
+
+  @Input() transaction: any;
+
+  @Input() type: string;
+
+  @Input() background: boolean;
+
+  constructor() { }
```
Comment (AoNoOokami) [inline projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.ts]: Can be removed if not used.

File: projects/admin/src/app/record/item-availability/item-availability.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  styles: []
+})
+export class ItemAvailabilityComponent implements OnInit {
+
+  @Input() item: any;
+
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Comment (AoNoOokami) [inline projects/admin/src/app/record/item-availability/item-availability.component.ts]: Can be removed if not used

Comment (Garfield-fr) [reply projects/admin/src/app/record/item-availability/item-availability.component.ts]: done

File: projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.ts]: done

File: projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      this.recordService
+        .getRecord('locations', record.metadata.location.pid)
+        .subscribe(location => {
+          this.location = location;
+          const libraryPid = extractIdOnRef(location.metadata.library.$ref);
+          this.recordService
+            .getRecord('libraries', libraryPid)
+            .subscribe(library => this.library = library);
+        });
+    });
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.ts]: I think you can refactor this to have only one subscribe, but It's not mandatory.

Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.ts]: done

File: projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    </div>
+    <div class="col-4" *ngIf="'loan_request' === type">
+      {{ transaction.metadata.pickup_location_pid | getRecord:'locations':'field':'pickup_name' | async }}
+    </div>
+    <div class="col-3" *ngIf="'loan_borrow' === type">
+      {{ transaction.metadata.end_date | dateTranslate:'longDate' }}
+    </div>
+    <div class="col-3" *ngIf="'loan_request' === type">
+        {{ transaction.metadata.transaction_date | dateTranslate:'longDate' }}
+      </div>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.html]: Cosmetic : Limit "IF" statement (my point of view)
```html
<ng-container *ngIf="type === 'loan_borrow'">
  <div class="col-4">
      {{ transaction.metadata.extension_count ?  transaction.metadata.extension_count : '-' }}
  </div>
  <div class="col-3">
      {{ transaction.metadata.end_date | dateTranslate:'longDate' }}
  </div>
</ng-container>
<ng-container *ngIf="type === 'loan_request'">
  <div class="col-4">
      {{ transaction.metadata.pickup_location_pid | getRecord:'locations':'field':'pickup_name' | async }}
  </div>
  <div class="col-3">
      {{ transaction.metadata.transaction_date | dateTranslate:'longDate' }}
  </div>
</ng-container>
```

Comment (Garfield-fr) [reply projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.html]: done

**Pull request merged**